### PR TITLE
harfbuzz: make `gobject-introspection` build-only

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/harfbuzz.rb [hunk 1]
```diff
@@ -16,12 +16,12 @@ class Harfbuzz < Formula
   end
 
   depends_on "glib-utils" => :build
+  depends_on "gobject-introspection" => :build
```
Comment (cho-m) [inline Formula/harfbuzz.rb:19]: Linux may be failing due to all Pythons (e.g. 3.10 from `meson`, 3.9 from `gobject-introspection`) being indirectly installed via build dependencies.

I guess any Python works to run build scripts like macOS Python 3.8

If we want to add test, a test dependency on either Python 3.9 or Python 3.10 could be used as `pygobject3` supports both.

**Comment by carlocab** [comment]
Can probably do the same with `gedit`, `gitg`, `libpeas`, `gjs`, `template-glib`. Maybe `pygobject3`, since Arch seems to only have it as a build dependency.

**Comment by cho-m** [comment]
> Can probably do the same with `gedit`, `gitg`, `libpeas`, `gjs`, `template-glib`. Maybe `pygobject3`, since Arch seems to only have it as a build dependency.

Arch actually has a split package with `gobject-introspection-runtime` for libraries https://archlinux.org/packages/extra/x86_64/gobject-introspection-runtime/

Based on Arch's dependents list, some of those formulae probably have linkage like `gitg`, `gjs`, and `pygobject3` (`python-gobject`)

**Comment by cho-m** [comment]
`ffms2` test failure seems to be unrelated and due to download site being down during run (sometime after ARM runners finished).

Not sure if CI will handle merge cleanly with #98809. In theory, there shouldn't be any merge conflicts. However, it may not matter as we already have a merge conflict with `mysql` in that PR so will need a rebase.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**